lib/promscrape/discovery/kubernetes/api_watcher.go

```diff
@@ -458,6 +464,11 @@ type urlWatcher struct {
 	apiURL    string
 	gw        *groupWatcher
 
+	refCount int
```

Comment:
can we add a comment reference of what we're counting here?

Reply:
Added a comment

Reply:
It actually tracks the number of register/unregister calls. Otherwise, you can simply take `len(aws)+len(awsPending)` instead.

Reply:
I think check for `len(aws)+len(awsPending)` is much simpler solution and must cover most cases.

Reply:
@f41gh7 could you please elaborate more on this statement of yours "Currently, another bug is possible. If api watcher resubscribes for cancelled urlWatcher, it'll receive an error."?
As I understand this was the reason why referenceCount was introduced. Could you please point me to the code where the mentioned by you bug could be reproduced?

Reply:
Sorry for delay. No it's not related to the reference count. Lets return back previous variant with if (len(uw.aws) + len(uw.awsPending)) == 0

 Previous implementation didn't remove entry from gw.m. Now it's fixed.

Reply:
@zekker6 can you update it and lets merge this fix.

Reply:
Updated